src/Query/FilterParser.ts

```diff
@@ -92,6 +92,12 @@ export function parseGrouper(line: string): Grouper | null {
                 return field.createGrouper();
             }
         }
+
+        // Temporary workaround for TagsField: query is done on plural 'tagS',
+        // but field.fieldNameSingular() returns 'tag'
+        if (line === 'group by tags' && field instanceof TagsField) {
+            return field.createGrouper();
+        }
```

Comment:
This comment made me look more closely at the change.

I adde a log statement in Grouper constructor and found that the property name for `group by tags` changed from `tags` to `tag`.

```ts
export class Grouper {
    public readonly property: string;
    public readonly grouper: GrouperFunction;

    constructor(property: string, grouper: GrouperFunction) {
        this.property = property;
        this.grouper = grouper;
        console.log(`${property}`);
    }
}
```

I guess that theoretically makes this a breaking change. What I don't know is whether it makes it a visible breaking change.

It's possible that the grouper name is used in the CSS.

Reply:
https://publish.obsidian.md/tasks/Advanced/Styling#Hidden+Components%2C+Groups+%26+Short+Mode

This suggests to me that `tags` would appear in the CSS:

![image](https://user-images.githubusercontent.com/4840096/231391237-273ed267-eb13-496c-9431-81eabf761f1e.png)

Reply:
Thanks a lot for noticing this.

I added a test which made me think whether this is a test that should be added to the other fields as well. We should not be testing only `TagsField` for this, aren't we?

I also made the workaround to be more generic ("less bad"). At first I tried to call the `field.fieldNamePlural()` but is not available here.

Reply:
> I added a test which made me think whether this is a test that should be added to the other fields as well. We should not be testing only `TagsField` for this, aren't we?

Thanks for the review! Just to finish this, do you think we need same test for other fields? Or it is fine as is

Reply:
> Thanks for the review! Just to finish this, do you think we need same test for other fields? Or it is fine as is

Thanks, I think it's fine as it is.